qiskit_ionq/ionq_backend.py

```diff
@@ -237,18 +271,78 @@ def _make_target(self) -> Target:
         tgt = Target(num_qubits=n)
 
         if self._gateset == "qis":
-            # 1q: H, S, T ; 2q: CX
-            for gate in (HGate(), SGate(), TGate()):
+            # 1-qubit (fixed)
+            for gate in (
+                IGate(),
+                XGate(),
+                ZGate(),
+                HGate(),
+                SGate(),
+                SdgGate(),
+                SXGate(),
+                SXdgGate(),
+                TGate(),
+                TdgGate(),
+            ):
                 tgt.add_instruction(gate, {(q,): None for q in range(n)})
+
+            # 1-qubit (parameterized)
+            theta, lam = Parameter("θ"), Parameter("λ")
+            for gate in (
+                RXGate(theta),
+                RYGate(theta),
+                RZGate(theta),
+                PhaseGate(lam),
+            ):
+                tgt.add_instruction(gate, {(q,): None for q in range(n)})
+
+            # 2-qubit (all-to-all)
             pairs = {(i, j): None for i in range(n) for j in range(n) if i != j}
-            tgt.add_instruction(CXGate(), pairs)
+
+            for gate in (
+                CXGate(),
+                CYGate(),
+                CZGate(),
+                CHGate(),
+                CSXGate(),
+                SwapGate(),
+            ):
+                tgt.add_instruction(gate, pairs)
+
+            twoq_param = (
+                CRXGate(theta),
+                CRYGate(theta),
+                CRZGate(theta),
+                CPhaseGate(lam),
+                RXXGate(theta),
+                RYYGate(theta),
+                RZZGate(theta),
+            )
+            for gate in twoq_param:
+                tgt.add_instruction(gate, pairs)
+
+            # all ordered, distinct k-tuples of qubits
+            def k_tuples(k: int) -> dict[tuple[int, ...], None]:
+                if n < k:
+                    return {}
+                return {t: None for t in itertools.permutations(range(n), k)}
```

Comment:
I'm afraid of k_tuples performance over time, but if it's not unreasonably slow right now this seems alright. Do we have to provide all possible qubits this way?

Reply:
no k_tuples is limited to k, so we only generate it for 4 qubits

Reply:
`k_tuples(4)` and `k_tuples(3)` lines

Reply:
right but a 4-tuple of the range over qubits, so it grows as the number of O(qubits^4). probably okay if its not perf-limiting now, but something to be aware of / keep an eye on